subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -954,6 +955,8 @@ private boolean isSettingsFileAvailable() {
 
         if (useOwnUserHomeServices || (gradleUserHomeDir != null && !gradleUserHomeDir.equals(buildContext.getGradleUserHomeDir()))) {
             properties.put(REUSE_USER_HOME_SERVICES, "false");
+        } else {
```

Comment:
❌ This looks wrong to me, setting the override for all tests that **don't** have their own user home dir. It should be the opposite.

Reply:
Why the opposite? If a test already has their own user home dir, its API jar cache is already private.

Reply:
Sorry, I got confused here. 

Another point though: Are you sure it's correct to skip this if `useOwnUserHomeServices` is true? I think it should have its own if-statement instead.